appIcons.js

```diff
@@ -747,6 +741,32 @@ const MyAppIcon = new Lang.Class({
         return backgroundColor;
     },
 
+    /**
+     * Downsample large icons before scanning for the backlight color to
+     * improve performance.
+     *
+     * @param pixBuf
+     * @param pixels
+     * @param resampleX
+     * @param resampleY
+     *
+     * @return [];
+     */
+    _resamplePixels: function (pixels, pixBuf, resampleX, resampleY) {
```

Comment:
Ow oops, good catch, I dont have any large icons so it was hard for me to test :D